### ISSUE TITLE
[Bug] Replacing MainPage with a NavigationPage does not work

### ISSUE THREAD

**Issue by humbertojaimes** [dotnet/maui#2517]
### Description

Replacing the MainPage using a NavigationPage does not work, it shows a blank screen

### Steps to Reproduce

1. Create a new app using the VS 2022 .NET MAUI template
2. Create a new BlankPage called Page2
3.  In the template default page change the button click event to replace the MainPage using a new NavigationPage - App.Current.MainPage = new NavigationPage(new Page2());
3. Execute the app

### Expected Behavior
The app replaces the MainPage with a new Page that supports navigation

### Actual Behavior
The app shows a page without controls. Replacing MainPage without NavigationPage creates the expected page 

### Basic Information

- Version with issue: 
    .NET MAUI in .NET 6 RC1
- IDE:
     Microsoft Visual Studio Community 2022 Preview (64-bit) Version 17.0.0 Preview 4.0 
- Platform Target Frameworks:
  - Android: API31

### Screenshots

![NavigationPage](https://user-images.githubusercontent.com/6464913/133618054-391180e0-879c-416c-baa6-58b30e0340e9.gif)

### Reproduction Link

[MauiApp2.zip](https://github.com/dotnet/maui/files/7177909/MauiApp2.zip)

### Workaround

Is not a work around but the NavigationPage works if the MainPage is set in App.xaml.cs ctor

**Comment by drasticactions** [comment]
I tried this on the newest main MAUI build, as I wanted to see if the Live Visual Tree would show the page or not when you reset it. For what it's worth, I couldn't reproduce it. However, I did find another issue when navigating back using the back button if the stack is empty. (https://github.com/dotnet/maui/issues/2521). It also didn't happen on iOS nor WinUI for me.

**Comment by humbertojaimes** [comment]
@drasticactions I tested it again and I can see the new page in the Live Visual Tree but not in the devices. 

![NavigationPage2](https://user-images.githubusercontent.com/6464913/133864395-ada7bcd1-b0ea-43d7-82a9-97c174665b9f.gif)

Both Windows and VS 2022 are fresh installations, and I use the API 29 and 30 Emulators and a Samsung Galaxy Tab 2 with Android 10. 

I tried the same project in WinUI and it works.
